static/app/stores/utils.tsx

```diff
@@ -0,0 +1,23 @@
+import {useEffect, useState} from 'react';
+import Reflux from 'reflux';
+
+type LegacyStoreShape = Reflux.Store & {
+  // Store must have `get` function that returns the current state
+  get: () => any;
+};
+
+/**
+ * Returns the state of a reflux store. Automatically unsubscribes when destroyed
+ *
+ * ```ts
+ * const teams = useLegacyStore(TeamStore);
+ * ```
+ */
+export function useLegacyStore<T extends LegacyStoreShape>(
```

Comment:
are we going to have more functions here in the future? otherwise, isn't better to rename the file to `useLegacyStore`?